pyfar/classes/audio.py

```diff
@@ -599,17 +621,21 @@ def __init__(
             self._n_samples = data.shape[-1]
             times = np.atleast_1d(
                 np.arange(0, self._n_samples) / sampling_rate)
-            TimeData.__init__(self, data, times, comment)
+            TimeData.__init__(self, data, times, comment, complex)
         elif domain == 'freq':
             # check and set n_samples
             if n_samples is None:
                 warnings.warn(
                     "Number of time samples not given, assuming an even "
                     "number of samples from the number of frequency bins.")
-                n_samples = (data.shape[-1] - 1)*2
+                if self._complex:
+                    n_samples = data.shape[-1]
+                else:
+                    n_samples = (data.shape[-1] - 1)*2
             elif n_samples > 2 * data.shape[-1] - 1:
```

Comment:
this needs a new if case depending on `self.complex`. Because n_samples can now also be n_bins